### PR TITLE
[WIP] Feature: pairwise edit distance for each string on a given nvstrings object 

### PR DESCRIPTION
**Pull request by kayush2O6**
This PR implements the `edit_distance_matrix()` function that computes pairwise distance on a list of strings(i.e. nvstrings object) and returns a distance matrix. 
Recently, I was working on a strings related use-case and where I felt the need of this function...

Example:
```python
import nvstrings, nvtext
s = nvstrings.to_device(["honda","hyundai","suzuki"])
n = nvtext.edit_distance_matrix(s)
print(n)
Out:    
[[0, 3, 6],
 [3, 0, 5],
 [6, 5, 0]]
```

### CODE REVIEW

File: cpp/src/text/edit_distance.cu [hunk 1]
```diff
@@ -227,3 +227,60 @@ unsigned int NVText::edit_distance( distance_type algo, NVStrings& strs1, NVStri
...
+    unsigned int* d_rtn = results;
+    if( !bdevmem )
+        d_rtn = device_alloc<unsigned int>(count*count,0);
+
+    // get the string pointers
+    rmm::device_vector<custring_view*> strings(count,nullptr);
+    custring_view** d_strings = strings.data().get();
+    strs.create_custring_index(d_strings);
+
+    for (unsigned int k = 0; k < count; k++)
```
Comment (davidwendt) [inline cpp/src/text/edit_distance.cu:250]: This is a CPU loop over the number of strings. Would recommend rewriting this routine to remove this CPU loop.

**Comment by davidwendt** [comment]
This repo is no longer active.

**Comment by kayush2O6** [comment]
ohh I see, I will submit this in `rapidsai/cudf` repo.

**Review by davidwendt — COMMENTED** [review]
> ohh I see, I will submit this in `rapidsai/cudf` repo.

I think that the C++/CUDA code here needs to be rewritten to remove the CPU loop. I can took a look at it next week.

File: cpp/src/text/edit_distance.cu (hunk 1 above)
Comment (kayush2O6) [reply cpp/src/text/edit_distance.cu:250]: Sure, I will work on it...

**Pull request closed without merging**